### PR TITLE
LPD-33047 Allow style books to be applied to a page based on the frontend token definition applied to the page.

### PR DESCRIPTION
**Pull request by algobob**
Ticket: https://liferay.atlassian.net/browse/LPD-33047

## Motivation
The user only applies a style book if the theme applied to a page is the same as the one applied to Public Pages. We want the user to be able to assign style books based on different token definitions to different pages of the same site.

## Proposed Solution
It filters the style books based on the frontend token definition applied to the page.

### CODE REVIEW

**Comment by algobob** [comment]
ci:test:sf

**Comment by algobob** [comment]
ci:test:sf

File: modules/apps/layout/layout-content-page-editor-web/src/main/java/com/liferay/layout/content/page/editor/web/internal/manager/StyleBookManager.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+		if (FeatureFlagManagerUtil.isEnabled(companyId, "LPD-30204")) {
+			FrontendTokenDefinition frontendTokenDefinition =
+				_frontendTokenDefinitionRegistry.getFrontendTokenDefinition(
+					companyId, layoutId, themeId);
+
+			return ListUtil.filter(
+				styleBookEntries,
+				styleBookEntry -> Objects.equals(
+					styleBookEntry.getThemeId(),
+					frontendTokenDefinition.getThemeId()));
```
Comment (thiago-buarqque) [inline modules/apps/layout/layout-content-page-editor-web/src/main/java/com/liferay/layout/content/page/editor/web/internal/manager/StyleBookManager.java:46]: Is it possible to `frontendTokenDefinition` be null?

File: modules/apps/layout/layout-content-page-editor-web/src/test/java/com/liferay/layout/content/page/editor/web/internal/manager/StyleBookManagerTest.java [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+		new AggregateTestRule(
+			CodeCoverageAssertor.INSTANCE, LiferayUnitTestRule.INSTANCE);
+
+	@Before
+	public void setUp() {
+		MockitoAnnotations.initMocks(this);
+	}
+
+	@Test
+	public void test() {
```
Comment (thiago-buarqque) [inline modules/apps/layout/layout-content-page-editor-web/src/test/java/com/liferay/layout/content/page/editor/web/internal/manager/StyleBookManagerTest.java:48]: This test can be simplified and cleaned to 

```java
FrontendTokenDefinition frontendTokenDefinition =
	Mockito.mock(FrontendTokenDefinition.class);

String dialectThemeId = RandomTestUtil.randomString();

Mockito.when(
	frontendTokenDefinition.getThemeId()
).thenReturn(
	dialectThemeId
);

Mockito.when(
	_frontendTokenDefinitionRegistry.getFrontendTokenDefinition(
		Mockito.anyLong(), Mockito.anyLong(), Mockito.anyString())
).thenReturn(
	frontendTokenDefinition
);

StyleBookEntry styleBookEntry1 = Mockito.mock(StyleBookEntry.class);
StyleBookEntry styleBookEntry2 = Mockito.mock(StyleBookEntry.class);
StyleBookEntry styleBookEntry3 = Mockito.mock(StyleBookEntry.class);

Mockito.when(
	styleBookEntry1.getThemeId()
).thenReturn(
	RandomTestUtil.randomString()
);

Mockito.when(
	styleBookEntry2.getThemeId()
).thenReturn(
	dialectThemeId
);

Mockito.when(
	styleBookEntry3.getThemeId()
).thenReturn(
	dialectThemeId
);

long groupId = RandomTestUtil.randomLong();

Mockito.when(
	_styleBookEntryLocalService.getStyleBookEntries(
		Mockito.eq(groupId), Mockito.anyInt(), Mockito.anyInt(),
		Mockito.any())
).thenReturn(
	Arrays.asList(styleBookEntry1, styleBookEntry2, styleBookEntry3)
);

List<StyleBookEntry> styleBooks =
	_styleBookManager.getStyleBooks(
		RandomTestUtil.randomLong(), groupId,
		RandomTestUtil.randomLong(), dialectThemeId);

Mockito.verify(
	_styleBookEntryLocalService
).getStyleBookEntries(
	Mockito.eq(groupId), Mockito.anyInt(), Mockito.anyInt(),
	Mockito.any()
);

Mockito.verify(
	_frontendTokenDefinitionRegistry
).getFrontendTokenDefinition(
	Mockito.anyLong(), Mockito.anyLong(), Mockito.anyString()
);

Assert.assertEquals(
	styleBooks.toString(), 2, styleBooks.size());

StyleBookEntry styleBookEntry4 = styleBooks.get(0);

Assert.assertEquals(dialectThemeId, styleBookEntry4.getThemeId());

styleBookEntry4 = styleBooks.get(1);

Assert.assertEquals(dialectThemeId, styleBookEntry4.getThemeId());
```

**Review by thiago-buarqque — CHANGES_REQUESTED** [review]
Thanks @algobob. Please remember to run baseline for the `frontend-token-definition-api`

**Comment by algobob** [comment]
> Thanks @algobob. Please remember to run baseline for the `frontend-token-definition-api`

I ran, but it didn't work :thinking: Let me try again

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties [hunk 3]
```diff
@@ -8914,6 +8914,7 @@ include-children-categories-when-searching-parent-categories-description=When ch
 include-clear-filters-option=Include Clear Filters Option
 include-decline-all-button=Include "Decline All" Button
 include-headers=Include Headers
+info-only-style-books-based-on-the-frontend-token-definition-provided-by-x-theme-are-visible="Info: Only Style Books based on the Frontend Token Definition provided by {0} Theme are visible."
```
[the remark this answers is not in the document]
Comment (algobob) [reply modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties]: > Why the double quotes?

My bad, I  automatically converted it as a string in my mind

File: modules/apps/layout/layout-content-page-editor-web/src/test/java/com/liferay/layout/content/page/editor/web/internal/manager/StyleBookManagerTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (algobob) [reply modules/apps/layout/layout-content-page-editor-web/src/test/java/com/liferay/layout/content/page/editor/web/internal/manager/StyleBookManagerTest.java:48]: > It seems to be because you forgot to enable the feature flag before the test runs

You're right. I added the feature flag as the last part of the story and forgot to re-test. Fixing it...

File: modules/apps/layout/layout-content-page-editor-web/src/main/java/com/liferay/layout/content/page/editor/web/internal/manager/StyleBookManager.java
[the remark this answers is not in the document]
Comment (algobob) [reply modules/apps/layout/layout-content-page-editor-web/src/main/java/com/liferay/layout/content/page/editor/web/internal/manager/StyleBookManager.java:1]: The idea was to make the design more testable. If I keep it in a private method, I would have to test through integration test, which was no the goal. There are others managers classes in the same package, so I just follow the design.

File: modules/apps/layout/layout-content-page-editor-web/src/test/java/com/liferay/layout/content/page/editor/web/internal/manager/StyleBookManagerTest.java [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+/**
+ * @author Anderson Luiz
+ */
+public class StyleBookManagerTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			CodeCoverageAssertor.INSTANCE, LiferayUnitTestRule.INSTANCE);
```
[the remark this answers is not in the document]
Comment (algobob) [reply modules/apps/layout/layout-content-page-editor-web/src/test/java/com/liferay/layout/content/page/editor/web/internal/manager/StyleBookManagerTest.java:40]: I'll remove this test rule for now and add it back when we remove the flag.

File: modules/apps/layout/layout-content-page-editor-web/src/main/java/com/liferay/layout/content/page/editor/web/internal/manager/StyleBookManager.java (hunk 1 above)
Comment (algobob) [reply modules/apps/layout/layout-content-page-editor-web/src/main/java/com/liferay/layout/content/page/editor/web/internal/manager/StyleBookManager.java:46]: I thought it wasn't, but it seems it can. Fixing it...

File: modules/apps/layout/layout-content-page-editor-web/src/test/java/com/liferay/layout/content/page/editor/web/internal/manager/StyleBookManagerTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (thiago-buarqque) [reply modules/apps/layout/layout-content-page-editor-web/src/test/java/com/liferay/layout/content/page/editor/web/internal/manager/StyleBookManagerTest.java:48]: >Too generic. It returns the style books based on the frontend token definition. Maybe testGetStyleBooksBaseOnFrontendTokenDefinition?

The test names are usually `test<MethodNameThatIsBeingTested>`.  If there are more than test scenario for the same method, it could be `test<MethodNameThatIsBeingTested>Scenario1` `test<MethodNameThatIsBeingTested>Scenario2` ...

**Comment by algobob** [comment]
It's stuck yet. I'll create a new pull request and link here soon.

**Comment by algobob** [comment]
Continues on #796
